### PR TITLE
fixed a bug in reading aerosol diagnostic outputs from GEFS-Aerosols

### LINKED ISSUE #677: Update the read interface for FV3 based model output in nemsio
The read interface for FV3 based model output in nemsio hasn't worked with gefs v12 atmos data since the last commit for changes of gefs v12.3 implementation. A fix is needed.

### PR DESCRIPTION
**Pull request by lipan-NOAA**
(https://github.com/NOAA-EMC/UPP/issues/677)

Some simulations (e.g. Atmosphere) do not have aerosol output in the surface flux file, so the code should handle this case gracefully. In this modification, we use "gocart_on" to enable this feature.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@lipan-NOAA Can you sync your branch with the latest UPP develop? The instructions can be found at
https://github.com/NOAA-EMC/UPP/wiki/UPP-Code-Development#update-your-branch-at-your-fork-with-branch-develop-at-authoritative-repository

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f [hunk 1]
```diff
@@ -237,7 +237,7 @@ SUBROUTINE INITPOST_GFS_NEMS_MPIIO(iostatusAER)
         end do
       end do
 
-!     write(*,*)'nrec=',nrec
+!     write(0,*)'nrec=',nrec
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: @lipan-NOAA Please revert this change. We would like the no-fatal messages go to stdout instead of stderr.

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f [hunk 2]
```diff
@@ -1156,7 +1156,7 @@ SUBROUTINE INITPOST_GFS_NEMS_MPIIO(iostatusAER)
                          vcrd, pint(1,j,lp1), psx2d(1,j), psy2d(1,j),   &
                          d2d,  u2d, v2d, pi2d, pm2d, omga2d, me)
 !     if (j ==1 .or. j == jm) &
-!     write (*,*)' omga2d=',omga2d(1,:),' j=',j
+!     write (0,*)' omga2d=',omga2d(1,:),' j=',j
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: @lipan-NOAA Please revert this change. We would like the no-fatal messages go to stdout instead of stderr.

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f [hunk 3]
```diff
@@ -1198,7 +1198,7 @@ SUBROUTINE INITPOST_GFS_NEMS_MPIIO(iostatusAER)
 !         Average omega for the last point near the poles - moorthi
           if (j ==1 .or. j == jm) then
             tx1 = 1.0 / im
-!     write(*,*)' j=',j,' omgamax=',maxval(omga(:,j,:)),' omgamin=',minval(omga(:,j,:))
+!     write(0,*)' j=',j,' omgamax=',maxval(omga(:,j,:)),' omgamin=',minval(omga(:,j,:))
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: @lipan-NOAA Please revert this change. We would like the no-fatal messages go to stdout instead of stderr.

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f [hunk 4]
```diff
@@ -1898,7 +1898,7 @@ SUBROUTINE INITPOST_GFS_NEMS_MPIIO(iostatusAER)
       do j=jsta,jend
         do i=1,im
           if (ths(i,j) /= spval) then
-!    write(*,*)' i=',i,' j=',j,' ths=',ths(i,j),' pint=',pint(i,j,lp1)
+!    write(0,*)' i=',i,' j=',j,' ths=',ths(i,j),' pint=',pint(i,j,lp1)
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: @lipan-NOAA Please revert this change. We would like the no-fatal messages go to stdout instead of stderr.

**Comment by WenMeng-NOAA** [comment]
@lipan-NOAA I tested your branch with gefs v12 atm model data files. Now the issue is fixed. You may find my test at  /u/wen.meng/ptmp/post_gefs_2023020200 on Cactus. Thanks!

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f (hunk 1 above)
Comment (lipan-NOAA) [reply sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: reverted it

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f (hunk 2 above)
Comment (lipan-NOAA) [reply sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: reverted it

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f (hunk 3 above)
Comment (lipan-NOAA) [reply sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: reverted it

File: sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f (hunk 4 above)
Comment (lipan-NOAA) [reply sorc/ncep_post.fd/INITPOST_GFS_NEMS_MPIIO.f]: reverted it

**Comment by lipan-NOAA** [comment]
I reverted those four changes as your suggested

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA Would you like to run the UPP RT tests on Hera and Orion with this PR?

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2 with no changed results.

**Comment by FernandoAndrade-NOAA** [comment]
UPP RT tests on Hera and Orion have completed with no changes to results.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.
